src/Command/Handler/StatusPaymentRequestHandler.php

```diff
@@ -0,0 +1,89 @@
+use PayPlug\SyliusPayPlugPlugin\Command\StatusPaymentRequest;
+use PayPlug\SyliusPayPlugPlugin\Handler\PaymentNotificationHandler;
+use Sylius\Abstraction\StateMachine\StateMachineInterface;
+use Sylius\Bundle\PaymentBundle\Provider\PaymentRequestProviderInterface;
+use Sylius\Component\Payment\Model\PaymentInterface;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Payment\PaymentRequestTransitions;
+use Sylius\Component\Payment\PaymentTransitions;
+use Symfony\Component\Messenger\Attribute\AsMessageHandler;
+
+#[AsMessageHandler]
+final class StatusPaymentRequestHandler
+{
+    public function __construct(
+        private PaymentRequestProviderInterface $paymentRequestProvider,
+        private StateMachineInterface $stateMachine,
+        private PayPlugApiClientFactoryInterface $apiClientFactory,
+        private PaymentNotificationHandler $paymentNotificationHandler,
+    ) {}
+
+    public function __invoke(StatusPaymentRequest $statusPaymentRequest): void
+    {
+        $paymentRequest = $this->paymentRequestProvider->provide($statusPaymentRequest);
+        $payment = $paymentRequest->getPayment();
+        if ('' !== $statusPaymentRequest->getForcedStatus()) {
+            $this->handleForcedStatus($statusPaymentRequest, $paymentRequest);
+            return;
+        }
+        // We don't have a forced status, so we retrieve the payment status from PayPlug
+        $client = $this->apiClientFactory->createForPaymentMethod($paymentRequest->getPayment()->getMethod());
+        $payplugPayment = $client->retrieve($payment->getDetails()['payment_id'] ?? throw new \LogicException('No PayPlug payment ID found in payment details.'));
+
+        $paymentRequest->setResponseData((array) $payplugPayment);
+        $details = new \ArrayObject($payment->getDetails());
+        $this->paymentNotificationHandler->treat($payment, $payplugPayment, $details);
+
+        $payment->setDetails($details->getArrayCopy());
+        $this->updatePaymentState($payment);
+
+        // Mark the PaymentRequest as completed
+        $this->stateMachine->apply(
+            $paymentRequest,
+            PaymentRequestTransitions::GRAPH,
+            PaymentRequestTransitions::TRANSITION_COMPLETE
+        );
+    }
+
+    private function handleForcedStatus(StatusPaymentRequest $statusPaymentRequest, PaymentRequestInterface $paymentRequest): void
+    {
+        $payment = $paymentRequest->getPayment();
+
+        $payment->setDetails([
+            ...$payment->getDetails(),
+            'status' => $statusPaymentRequest->getForcedStatus(),
+        ]);
+
+        $this->updatePaymentState($payment);
+
+        // Mark the PaymentRequest as completed
+        $this->stateMachine->apply(
+            $paymentRequest,
+            PaymentRequestTransitions::GRAPH,
+            PaymentRequestTransitions::TRANSITION_COMPLETE
+        );
+    }
+
+    private function updatePaymentState(PaymentInterface $payment): void
```

Comment:
Duplicate of `NotifyPaymentRequestHandler::updatePaymentState`, should we refactor it?

Reply:
Let's keep like that for now. We may have to come back on these developments. On a next iteration, we could add a new service to apply state machine change